nifi-nar-bundles/nifi-marklogic-bundle/nifi-marklogic-processors/src/main/java/org/apache/nifi/marklogic/processor/ExtensionCallMarkLogic.java

```diff
@@ -214,12 +214,14 @@ public void onTrigger(ProcessContext context, ProcessSession session) throws Pro
             if (format != null) {
                 bytesHandle.withFormat(Format.valueOf(format));
             }
-            synchronized(resourceManager) {
-                ServiceResultIterator resultIterator = resourceManager.callService(method, bytesHandle, requestParameters);
-                while(resultIterator.hasNext()) {
-                    ServiceResult result = resultIterator.next();
-                    session.append(flowFile, out -> out.write(result.getContent(new BytesHandle()).get()));
-                }
+
+            ServiceResultIterator resultIterator = resourceManager.callService(method, bytesHandle, requestParameters);
+            while(resultIterator.hasNext()) {
+                ServiceResult result = resultIterator.next();
+                session.append(flowFile, out -> out.write(result.getContent(new BytesHandle()).get()));
+            }
+
+            synchronized(session) {
```

Comment:
I modified this PR to only pertain to fixing where synchronized is used - it's now consistent with every other processor.

I don't know of any good way to write a test for this however - though we don't have tests for synchronized being used anywhere else. I would like to add a protected method to AbstractMarkLogicProcessor to handle this, which would also document why it's needed (apparently when lots of FlowFiles are being created, problems can arise when the transfer/commit calls on the session are not synchronized).